File: rs/hang/src/catalog/container.rs
use serde::{Deserialize, Serialize};

/// Container format for frame timestamp encoding and frame payload structure.
///
/// - "native": Uses QUIC VarInt encoding (1-8 bytes, variable length), raw frame payloads
/// - "raw": Uses fixed u64 encoding (8 bytes, big-endian), raw frame payloads  
/// - "cmaf": Fragmented MP4 container - frames contain complete moof+mdat fragments
#[derive(Serialize, Deserialize, Debug, Clone, Copy, PartialEq, Eq, Default)]
#[serde(rename_all = "camelCase")]
pub enum Container {
	#[serde(rename = "native")]
	#[default]
	Native,
	#[serde(rename = "raw")]
	Raw,
	#[serde(rename = "cmaf")]
	Cmaf,
}

File: js/hang/src/catalog/container.ts
import { z } from "zod";

/**
 * Container format for frame timestamp encoding.
 *
 * - "native": Uses QUIC VarInt encoding (1-8 bytes, variable length)
 * - "raw": Uses fixed u64 encoding (8 bytes, big-endian)
 * - "cmaf": Fragmented MP4 container (future)
 */
export const ContainerSchema = z.enum(["native", "raw", "cmaf"]);

export type Container = z.infer<typeof ContainerSchema>;

/**
 * Default container format when not specified.
 * Set to native for backward compatibility.
 */
export const DEFAULT_CONTAINER: Container = "native";
